File: IOModels/UserModel.cs
namespace SmartAlerts.API.IOModels
{
    public class UserModel
    {
        /// <summary>
        /// Gets or Sets the User Id
        /// </summary>
        public long UserId { get; set; }
        /// <summary>
        /// First name of the user.
        /// </summary>
        public required string FirstName { get; set; }

        /// <summary>
        /// Last name of the user.
        /// </summary>
        public required string LastName { get; set; }

        /// <summary>
        /// Email address of the user.
        /// </summary>
        public required string Email { get; set; }

        public long UserTypeId { get; set; }

        /// <summary>
        /// Name of the user's shop.
        /// </summary>
        public string? ShopName { get; set; }

        /// <summary>
        /// First line of the user's address.
        /// </summary>
        public required string Address1 { get; set; }

        /// <summary>
        /// Second line of the user's address.
        /// </summary>
        public string? Address2 { get; set; }

        /// <summary>
        /// City of the user's address.
        /// </summary>
        public required string City { get; set; }

        /// <summary>
        /// Region or state of the user's address.
        /// </summary>
        public required string Region { get; set; }

        /// <summary>
        /// Postal code of the user's address.
        /// </summary>
        public required string PostalCode { get; set; }

        /// <summary>
        /// Country of the user's address.
        /// </summary>
        public required string Country { get; set; }

        /// <summary>
        /// Phone number of the user.
        /// </summary>
        public required string Phone { get; set; }

        public required string? ProfilePicUrl { get; set; }
    }
}

File: main/java/com/smart/alerts/models/UserModel.java
package com.smart.alerts.models;

public class UserModel {

    public long UserId;
    public String FirstName;
    public String LastName;
    public String Email;
    public long UserTypeId;
    public String ShopName;
    public String Address1;
    public String Address2;
    public String City;
    public String Region;
    public String PostalCode;
    public String Country;
    public String Phone;
    public String ProfilePicUrl;

    public UserModel() {
        UserId = 0;
        FirstName = "";
        LastName = "";
        Email = "";
        UserTypeId = 0;
        ShopName = "";
        Address1 = "";
        Address2 = "";
        City = "";
        Region = "";
        PostalCode = "";
        Country = "";
        Phone = "";
        ProfilePicUrl = "";
    }
}
